subprojects/core/src/integTest/groovy/org/gradle/normalization/ConfigureRuntimeClasspathNormalizationIntegrationTest.groovy

```diff
@@ -466,16 +472,56 @@ class ConfigureRuntimeClasspathNormalizationIntegrationTest extends AbstractInte
         executedAndNotSkipped(project.customTask)
     }
 
+    @ToBeFixedForConfigurationCache(because = "classpath normalization")
+    @Issue('https://github.com/gradle/gradle/issues/16144')
+    def "changing normalization configuration rules changes build cache key (#description)"() {
+        def project = new ProjectWithRuntimeClasspathNormalization(Api.RUNTIME)
+        project.propertiesFileInJar.changeProperty(ALSO_IGNORE_ME, 'some value')
+        project.buildFile << """
+            normalization {
+                runtimeClasspath {
+                    if (project.hasProperty('${enableFilterFlag}')) {
```

Comment:
We implement it this way, rather than just changing the build script, because we don't want the change in build script to affect the cache hit.  We want the _only_ change to be in the normalization rules so we can be sure that's what's changing the cache key.

Reply:
I think that comment belongs to the test code.

Reply:
Good point.  Done.